### ISSUE TITLE
dereference.of.nullable when null not possible

### ISSUE THREAD

**Issue by scottkennedy** [typetools/checker-framework#728]
I'm seeing this error:
```
File.java:794: error: [dereference.of.nullable] dereference of possibly-null reference endAction
                        endAction.run();
                        ^
```

With this code:
```java
private static void method(@NonNull View view, @Nullable final Runnable endAction) {
    ViewPropertyAnimator animator = view.animate();

    if (endAction != null) {
        animator.setListener(new AnimatorListener() {
            @Override
            public void onAnimationEnd(Animator animation) {
                super.onAnimationEnd(animation);
                endAction.run(); // This is the line mentioned
            }
        });
    }

    animator.start();
}
```

I'm pretty sure it's impossible for `endAction` to be `null` here.

**Comment by wmdietl** [comment]
Thanks for this report!
We have been tracking this as Issue #266 .
Comment https://github.com/typetools/checker-framework/issues/266#issuecomment-118318478 mentions a simple work-around.
If this issue arises frequently for your code, we should re-evaluate its priority.
